src/models/model.cpp

```diff
@@ -437,7 +437,24 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
     } else if (provider_options.name == "dml") {
       if (!p_dml_api_) {
         auto current_module_path = CurrentModulePath();
-        dml_objects_ = DmlHelpers::CreateDmlObjects(current_module_path);
+
+        bool contains_device_luid = false;
+        LUID device_luid{};
+        for (const auto& [name, value] : provider_options.options) {
+          if (name == "luid_high_part") {
+            device_luid.HighPart = std::stol(value);
+            contains_device_luid = true;
+          } else if (name == "luid_low_part") {
+            device_luid.LowPart = std::stol(value);
+            contains_device_luid = true;
+          }
```

Comment:
What happens if the user only specifies the low or high part of the luid? Alternatively, can't we simply parse the entire LUID from a single string?

Reply:
If the user only specifies the low or high part of the luid, the program will crush. Yes, I can change it and specify luid by a single string, where high part and low part will be separated by ":" symbol

Reply:
Updated